Unable to run the Module Federation host server with the Webpack bundle ( --bundler=webpack).

### Current Behavior

According to https://nx.dev/recipes/module-federation/create-a-host guide, after generating the host with:
`nx g @nx/react:host apps/react/shell --bundler=webpack`

and trying to run:
`nx serve shell`

it returns:
NX   Failed to process project graph.
An error occurred while processing files for the @nx/webpack/plugin plugin

### Expected Behavior

It should serve as it does with the Rspack bundler.


### GitHub Repo

_No response_

### Steps to Reproduce

1. Install new project 
`npx create-nx-workspace@latest <project-name>`

![Image](https://github.com/user-attachments/assets/dc55363a-59bb-4d6e-bdb7-23d9b8972fac)

2. Generate module federation host, use webpack instead of rspack:
nx g @nx/react:host apps/react/shell --bundler=webpack

3. run `nx serve shell`

### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node           : 22.14.0
OS             : win32-x64
Native Target  : x86_64-windows
npm            : 10.9.2

nx (global)            : 20.5.0
nx                     : 20.5.0
@nx/js                 : 20.5.0
@nx/jest               : 20.5.0
@nx/eslint             : 20.5.0
@nx/workspace          : 20.5.0
@nx/devkit             : 20.5.0
@nx/eslint-plugin      : 20.5.0
@nx/module-federation  : 20.5.0
@nx/react              : 20.5.0
@nx/web                : 20.5.0
@nx/webpack            : 20.5.0
typescript             : 5.7.3
---------------------------------------
Registered Plugins:
@nx/webpack/plugin
@nx/eslint/plugin
@nx/jest/plugin
---------------------------------------
⚠️ Unable to construct project graph.
Failed to process project graph.
     - apps/react/shell/webpack.config.ts: Error: Cannot find module '@rspack/core'
  Require stack:
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\src\utils\share.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\src\utils\index.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\src\with-module-federation\webpack\utils.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\src\with-module-federation\webpack\with-module-federation.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\webpack.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\apps\react\shell\webpack.config.ts
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\webpack\src\utils\webpack\resolve-user-defined-webpack-config.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\webpack\src\plugins\plugin.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\webpack\plugin.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\nx\src\project-graph\plugins\load-resolved-plugin.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\nx\src\project-graph\plugins\isolation\plugin-worker.js
      at Function.<anonymous> (node:internal/modules/cjs/loader:1225:15)
      at Function.Module._resolveFilename (C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\tsconfig-paths\src\register.ts:115:36)
      at Function._load (node:internal/modules/cjs/loader:1055:27)
      at TracingChannel.traceSync (node:diagnostics_channel:322:14)
      at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
      at Module.require (node:internal/modules/cjs/loader:1311:12)
      at require (node:internal/modules/helpers:136:16)
      at Object.<anonymous> (C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\src\utils\share.js:14:16)
      at Module._compile (node:internal/modules/cjs/loader:1554:14)
      at node:internal/modules/cjs/loader:1706:10
```

### Failure Logs

```shell
⚠️ Unable to construct project graph.
Failed to process project graph.
     - apps/react/shell/webpack.config.ts: Error: Cannot find module '@rspack/core'
  Require stack:
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\src\utils\share.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\src\utils\index.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\src\with-module-federation\webpack\utils.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\src\with-module-federation\webpack\with-module-federation.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\webpack.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\apps\react\shell\webpack.config.ts
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\webpack\src\utils\webpack\resolve-user-defined-webpack-config.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\webpack\src\plugins\plugin.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\webpack\plugin.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\nx\src\project-graph\plugins\load-resolved-plugin.js
  - C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\nx\src\project-graph\plugins\isolation\plugin-worker.js
      at Function.<anonymous> (node:internal/modules/cjs/loader:1225:15)
      at Function.Module._resolveFilename (C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\tsconfig-paths\src\register.ts:115:36)
      at Function._load (node:internal/modules/cjs/loader:1055:27)
      at TracingChannel.traceSync (node:diagnostics_channel:322:14)
      at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
      at Module.require (node:internal/modules/cjs/loader:1311:12)
      at require (node:internal/modules/helpers:136:16)
      at Object.<anonymous> (C:\Users\ferra\Desktop\wa\NX-DEV\nx-module-federation\node_modules\@nx\module-federation\src\utils\share.js:14:16)
      at Module._compile (node:internal/modules/cjs/loader:1554:14)
      at node:internal/modules/cjs/loader:1706:10
```

### Package Manager Version

_No response_

### Operating System

- [ ] macOS
- [ ] Linux
- [x] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

Happens with Angular builds as well. Created a host and remote in an integrated monorepo using the command below and then just ran it

`nx g @nx/angular:host apps/host --remotes=myremote --standalone`

`nx run host:serve:development`

 NX   Cannot find module '@rspack/core'

Require stack:
- /Users/---/Development/testmfe/nx-bug/node_modules/@nx/module-federation/src/utils/share.js
- /Users/---/Development/testmfe/nx-bug/node_modules/@nx/module-federation/src/utils/index.js
- /Users/---/Development/testmfe/nx-bug/node_modules/@nx/module-federation/src/executors/utils/start-remote-iterators.js
...

--- a/packages/module-federation/package.json
+++ b/packages/module-federation/package.json
@@ -30,14 +30,12 @@
     "@nx/web": "file:../web",
     "picocolors": "^1.1.0",
     "webpack": "^5.88.0",
+    "@rspack/core": "^1.1.5",
     "@module-federation/enhanced": "^0.9.0",
     "@module-federation/node": "^2.6.26",
     "@module-federation/sdk": "^0.9.0",
     "express": "^4.21.2",
     "http-proxy-middleware": "^3.0.3"
-  },
-  "peerDependencies": {
-    "@rspack/core": "^1.1.5"
   },
   "nx-migrations": {
     "migrations": "./migrations.json"

